hazelcast/src/main/java/com/hazelcast/query/impl/getters/AbstractJsonGetter.java

```diff
@@ -113,6 +143,42 @@ Object getValue(Object obj, String attributePath) throws Exception {
         }
     }
 
+    @Override
+    Object getValue(Object obj, String attributePath, Object metadata) throws Exception {
+        if (metadata == null) {
+            return getValue(obj, attributePath);
+        }
+        JsonSchemaNode schemaNode = (JsonSchemaNode) metadata;
+
+        NavigableJsonInputAdapter adapter = annotate(obj);
+        JsonGetterContext queryContext = contextCache.getContext(attributePath);
+        List<WeightedItem<JsonPattern>> patternsSnapshot = queryContext.getPatternListSnapshot();
+
+        JsonPathCursor pathCursor = queryContext.newJsonPathCursor();
+        JsonPattern knownPattern = null;
+        for (int i = 0; i < patternTryCount && i < patternsSnapshot.size(); i++) {
```

Comment:
If we always iterate to less than `patternTryCount`, does the `queryContext.getPatternListSnapshot()` ever need to be greater than the `patternTryCount`?

Reply:
We need to keep track of observed frequency of more patterns than we try. Otherwise less frequent patterns cannot get more votes than 1. Then there is "no less frequent pattern" to come and beat "more frequent patterns" in time. In the end, we stick with with the first `patternTryCount` patterns we saw for the entire lifetime of the query engine.

Reply:
Maybe I'm missing something but let's say there are five patterns `[p1, p2, p3, p4, p5]`. If we always break the loop with `i < patternTryCount`, how do p4 and p5 get tried? And if we break the loop after three patterns, without trying p4 and p5, can it happen that we continue to create a pattern which is equal to p4 or p5 and add it again to the list (e.g. `[p1, p2, p3, p4, p5, p4]`)?

Maybe I'm just missing something.

Reply:
I see your point. My intention was to increase the weight of a less frequently used value if we try to re-add it to the list instead of duplicating. Clearly, it does not work the way I intented. Requires a fix. Thanks